matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/timeline/LoadTimelineStrategy.kt

```diff
@@ -312,3 +347,40 @@ internal class LoadTimelineStrategy constructor(
         }
     }
 }
+
+/**
+ * Helper to observe and query the live room state.
+ */
+private class LiveRoomStateListener(
```

Comment:
I understand it's nice to have private class, but we try to also have one class per file, and keep files small.
So I would vote to extract this to its own file.
The class could be `internal`.